### PR TITLE
Introduce SendComponent

### PR DESCRIPTION
**Pull request by timbussmann**

### CODE REVIEW

**Comment by timbussmann** [comment]
This one is good to go I think. Note that `IMessageSession` is currently not provided by this component as it mostly depends on the pipeline components root context logic. I'm currently looking into resolving that but that's better suited for a dedicated PR.

File: src/NServiceBus.Core/Pipeline/Outgoing/SendComponent.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
+﻿namespace NServiceBus.Pipeline.Outgoing
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Pipeline/Outgoing/SendComponent.cs]: This should be just `NServiceBus`

File: src/NServiceBus.Core/Pipeline/Outgoing/SendComponent.cs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        }
+
+        public static SendComponent Initialize(Configuration configuration, PipelineSettings pipelineSettings, HostingComponent hostingComponent, RoutingComponent routingComponent)
+        {
+            var newIdGenerator = GetIdStrategy(configuration);
+            pipelineSettings.Register("AttachCausationHeaders", new AttachCausationHeadersBehavior(newIdGenerator), "Adds related to and conversation id headers to outgoing messages");
+
+            pipelineSettings.Register(new AttachSenderRelatedInfoOnMessageBehavior(), "Makes sure that outgoing messages contains relevant info on the sending endpoint.");
+            pipelineSettings.Register("AuditHostInformation", new AuditHostInformationBehavior(hostingComponent.HostInformation, hostingComponent.EndpointName), "Adds audit host information");
+            pipelineSettings.Register("AddHostInfoHeaders", new AddHostInfoHeadersBehavior(hostingComponent.HostInformation, hostingComponent.EndpointName), "Adds host info headers to outgoing headers");
```
Comment (SeanFeldman) [inline src/NServiceBus.Core/Pipeline/Outgoing/SendComponent.cs]: It's odd to see host information related behaviors moved from the `HostingComponent` here.
What am I missing?

Comment (timbussmann) [reply src/NServiceBus.Core/Pipeline/Outgoing/SendComponent.cs]: the host doesn't care where it's information are being used. Especially it doesn't have a requirement to add these information to outgoing messages. This is a decision made by the outgoing pipeline.

**Pull request merged**